source/common/filesystem/posix/directory_iterator_impl.cc

```diff
@@ -55,7 +55,11 @@ FileType DirectoryIteratorImpl::fileType(const std::string& full_path) const {
 
   const Api::SysCallIntResult result = os_sys_calls_.stat(full_path.c_str(), &stat_buf);
   if (result.rc_ != 0) {
-    throw EnvoyException(fmt::format("unable to stat file: '{}'", full_path));
+    if (errno == ENOENT)
+      // Special case, a symlink is broken, target couldn't be stat'ed
```

Comment:
Sorry I'm not clear on what this means. Can you flesh out this comment a bit more?